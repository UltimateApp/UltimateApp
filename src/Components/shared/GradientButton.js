import React from 'react';
import { StyleSheet, Text, TouchableOpacity } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';

import theme from '../../styles/theme.style';

const GradientButton = props => {
  return (
    <TouchableOpacity style={styles.button} onPress={props.onPress} testID="gradientButton">
      <LinearGradient
        style={styles.gradient}
        colors={[theme.GRADIENT_FIRST_COLOR, theme.GRADIENT_SECOND_COLOR]}
        start={{ x: 1, y: 1 }}
        end={{ x: 0, y: 0 }}
      >
        <Text style={styles.text}>{props.text}</Text>
      </LinearGradient>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    margin: 20,
    width: 100,
    height: 100,
    borderRadius: 50,
    justifyContent: 'center',
    alignItems: 'center',
  },
  gradient: {
    width: 100,
    height: 100,
    borderRadius: 50,
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    fontSize: theme.FONT_SIZE_MEDIUM,
    fontWeight: 'bold',
  },
});

export default GradientButton;
