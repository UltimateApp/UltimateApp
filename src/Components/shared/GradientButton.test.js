import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, cleanup } from 'react-native-testing-library';

import GradientButton from './GradientButton';

afterEach(cleanup);

describe('<GradientButton />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<GradientButton text="OK" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls delegates onPress prop', async () => {
    const onPress = jest.fn();

    const { getByTestId } = render(<GradientButton text="Easy" onPress={onPress} />);

    await fireEvent.press(getByTestId('gradientButton'));

    expect(onPress).toHaveBeenCalled();
  });
});
