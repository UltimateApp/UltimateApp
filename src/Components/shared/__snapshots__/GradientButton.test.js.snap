// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GradientButton /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 50,
      "height": 100,
      "justifyContent": "center",
      "margin": 20,
      "opacity": 1,
      "width": 100,
    }
  }
  testID="gradientButton"
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          4278759146,
          4281005464,
        ],
        "endPoint": Array [
          0,
          0,
        ],
        "locations": undefined,
        "startPoint": Array [
          1,
          1,
        ],
      }
    }
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 50,
        "height": 100,
        "justifyContent": "center",
        "width": 100,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      OK
    </Text>
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;
